lightly/api/download.py

```diff
@@ -366,3 +316,82 @@ def download_prediction_file(
         return None # the file doesn't exist!
 
     return response.json()
+
+def download_video_frames_at_timestamps(
+        url: str,
+        timestamps: List[int],
+        as_pil_image: int = True,
+        thread_type: av.codec.context.ThreadType = av.codec.context.ThreadType.AUTO,
+        video_channel: int = 0,
+    ) -> Iterable[Union[PIL.Image.Image, av.VideoFrame]]:
+        """Lazily retrieves frames from a video at a specific timestamp stored at the given url.
+
+        Args:
+            url:
+                The url where video is downloaded from.
+            timestamps:
+                Timestamps in pts from the start of the video. The images
+                at these timestamps are returned.
+                The timestamps must be strictly monotonically ascending.
+                See https://pyav.org/docs/develop/api/time.html#time
+                for details on pts.
+            as_pil_image:
+                Whether to return the frame as PIL.Image.
+            thread_type:
+                Which multithreading method to use for decoding the video.
+                See https://pyav.org/docs/stable/api/codec.html#av.codec.context.ThreadType
+                for details.
+            video_channel:
+                The video channel from which frames are loaded.
+
+        Returns:
+            A generator that loads and returns a single frame per step.
+
+        """
+        _check_av_available()
+
```

Comment:
I think you have to add a test here if `timestamps` is empty and return an empty iterator in this case. Otherwise we will get exceptions later on in the function.

Reply:
I added that test/fix and the according unittest.